src/responder/common/responder_common.c

```diff
@@ -262,8 +264,8 @@ errno_t csv_string_to_uid_array(TALLOC_CTX *mem_ctx, const char *csv_string,
     ret = EOK;
 
 done:
-    if(setenv("_SSS_LOOPS", "NO", 0) != 0) {
-        DEBUG(SSSDBG_OP_FAILURE, "Failed to set _SSS_LOOPS.\n");
+    if (setenv("_SSS_LOOPS", loops_were_allowed ? "YES" : "NO" , 1) != 0) {
```

Comment:
This will make code like `setenv("_SSS_LOOPS", "NO", 0)` to fail later.
So you either need to update 3rd arg to 1 in all other invocations of `setenv("_SSS_LOOPS"...) or keep existing pattern with `unsetenv`.
I would prefer the latter.

Reply:
Did the latter.